test/evergreen/run_format_configs.sh

```diff
@@ -21,20 +20,103 @@ fi
 
 success=0
 failure=0
+running=0
+parallel_jobs=8
+cleanup=0
+PID=""
+declare -a PID_LIST
+
+usage() {
+	echo "usage: $0 "
+	echo "    -j parallel  jobs to execute in parallel (defaults to 8)"
+
+	exit 1
+}
+
+
+while :; do
+	case "$1" in
+	-j)
+		parallel_jobs="$2"
+		[[ "$parallel_jobs" =~ ^[1-9][0-9]*$ ]] ||
+			fatal_msg "-j option argument must be a non-zero integer"
+		shift ; shift ;;
+	-*)
+		usage ;;
+	*)
+		break ;;
+	esac
+done
+
+# Wait for other format runs.
+wait_for_process()
+{
+	while true
+	do
+		for process in ${PID_LIST[@]};do
+			if [ ps $process > /dev/null ] ; then
+				# The process id is running so sleep for 2 second before checking another
+				# process status.
+				sleep 2
```

Comment:
I did not fully get the point of why sleep here if one process is running before checking another process. Should the sleep be put down outside this for loop (i.e. avoid delay checking for other processes that could have finished)?